mbuild/tests/test_compound.py

```diff
@@ -1332,6 +1332,157 @@ def test_none_charge(self):
     def test_energy_minimize(self, octane):
         octane.energy_minimize()
 
+    @pytest.mark.skipif(
+        not has_openbabel, reason="Open Babel package not installed"
+    )
+    def test_energy_minimize_shift_com(self, octane):
+        com_old = octane.pos
+        octane.energy_minimize()
+        # check to see if COM of energy minimized Compound
+        # has been shifted back to the original COM
+        assert np.allclose(com_old, octane.pos)
+
+    @pytest.mark.skipif(
+        not has_openbabel, reason="Open Babel package not installed"
+    )
+    def test_energy_minimize_shift_anchor(self, octane):
+        anchor_compound = octane.labels["chain"].labels["CH3"][0]
+        pos_old = anchor_compound.pos
+        octane.energy_minimize(anchor=anchor_compound)
+        # check to see if COM of the anchor Compound
+        # has been shifted back to the original COM
+        assert np.allclose(pos_old, anchor_compound.pos)
+
+    @pytest.mark.skipif(
+        not has_openbabel, reason="Open Babel package not installed"
+    )
+    def test_energy_minimize_fix_compounds(self, octane):
+        methyl_end0 = octane.labels["chain"].labels["CH3"][0]
+        methyl_end1 = octane.labels["chain"].labels["CH3"][1]
+        carbon_end = octane.labels["chain"].labels["CH3"][0].labels["C"][0]
+        not_in_compound = mb.Compound(name="H")
+
+        # fix the whole molecule and make sure positions are close
+        # given stochastic nature and use of restraining springs
+        # we need to have a pretty loose tolerance for checking
+        old_com = octane.pos
+        octane.energy_minimize(
+            fixed_compounds=octane, shift_com=False, constraint_factor=1e6
+        )
+        assert np.allclose(octane.pos, old_com, rtol=1e-2, atol=1e-2)
+
+        # primarily focus on checking inputs are parsed correctly
+        octane.energy_minimize(fixed_compounds=[octane])
+        octane.energy_minimize(fixed_compounds=carbon_end)
+        octane.energy_minimize(fixed_compounds=methyl_end0)
+        octane.energy_minimize(fixed_compounds=[methyl_end0])
+        octane.energy_minimize(
+            fixed_compounds=[methyl_end0, (True, True, True)]
+        )
+        octane.energy_minimize(
```

Comment:
I might expect based on the if elif cascade mentioned in line 2494 of compound.py, that this would only add the fixed_position restraint in the X dimension, and not in the Y.

We might want to add more specific tests to make sure those restraints are actually being satisfied, because as of now I'm not entirely sure.

Reply:
oh, good catch. I fixed up the logic on that one.